oeps/architectural-decisions/oep-0065/decisions/0002-frontend-app-migrations.rst

```diff
@@ -0,0 +1,120 @@
+  - ``dev`` will start a dev server, loading the repository's application modules into the shell in a site.
+  - ``dev:module`` will start a dev server that provides the application modules via module federation.
+  - ``build`` will create a standalone deployable artifact that uses the shell (similar to the micro-frontend architecture)
+  - ``build:module`` will create a standalone deployable artifact that provides the application modules via module federation.
+  - ``release`` will package the library for distribution on npm.
+  - ``serve`` will work with ``build`` or ``build:module`` to locally serve the production assets they generated.
+  - ``pack`` will work with ``release`` to create a ``.tgz`` file suitable for installing in local git checkouts that depend on the library.  (this is a development tool)
+- The ``dev``, ``dev:module``, ``build``, and ``build:module`` CLI commands will rely on the existence of a "Site Config" file (the replacement for .env/env.config files) which will not be checked into the repository.
+- frontend-app-* repositories that are part of Open edX releases will be expected to be published on NPM as a library which exports its modules.
+
+Redefining "application"
+========================
+
+This decision constitutes a re-definition of "application" in the Open edX frontend ecosystem.  In the past, "application" was synonymous with the entire frontend-app-* repository.  With this ADR, "application" will be synonymous with the top-level modules provided by a frontend-app-* repository. Those modules are applications.
+
+We believe that the boundaries of our frontend-app-* repositories are fairly arbitrary and not as semantically meaningful as the boundaries of the *application modules* within them.  We want these modules to be the primary unit of modularity - not the entire repository.  The shell will be configured to load applications, not repositories.
+
+As an example:
+
+* The ``frontend-app-learning`` repository contains modules related to "learning" - course outline, courseware, the progress and dates pages, etc.
+* The ``frontend-app-authn`` repository contains modules related to "authentication" - login, registration, logout, forgot password, etc.
+
+The shell will be configured to load things like "courseware" or "login", not "learning" and "authn" - the repository the module is in is incidental.
+
+Regarding "plugins"
+-------------------
+
+Plugins are another type of module.  We believe that they are different than application modules because they have a different interface with the shell.  Plugins are generally UI widgets loaded into a slot, though some can be larger.  Applications are generally large and complex, comprising multiple pages of functionality with internal routing.  They are also loaded at a path in the application, not in a plugin slot.
```

Comment:
If my thoughts from the "redefining application" section make sense, then I think this could be changed too.

Instead of
* Redefining "application"
* Regarding "plugins"

We could have
* Thinking in modules
  * Application modules
  * Plugin modules

Reply:
I do like this reframing to "thinking in modules".  I just think the phrase "micro-frontend application" is so ubiquitous right now that we explicitly need to say we're changing the definition, you know?  I don't want folks to be confused because we didn't call that out. 🤔  The repos are literally called "frontend apps", and this is a re-definition of what that means.  I don't like that the repo name doesn't quite feel right anymore either, but I don't feel like we can go around and rename them to frontend-library-learning, etc. 🙁

Reply:
@brian-smith-tcril I took another stab at this entire section in the spirit of your comments here.  I think it's all better.